### PR TITLE
Updated to Operator SDK 0.5.0

### LINKED ISSUE #272: Update to Operator SDK 0.5.0
The Operator SDK 0.5.0 was released almost a month ago and contains some OLM-related commands that we could make use of.

### PR DESCRIPTION
**Pull request by jpkrohling**
Closes #272

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/273)

File: deploy/operator.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
         - name: jaeger-operator
           image: jaegertracing/jaeger-operator:1.10.0
           ports:
-          - containerPort: 60000
+          - containerPort: 8383
```
Comment (objectiser) [inline deploy/operator.yaml]: Does this port have any particular significance? Would it be better to have something in the same ballpark as other Jaeger ports?

**Comment by pavolloffay** [comment]
@jpkrohling could you please make sure that self-provisioning of ES cluster works? You mentioned that this one generates more one additional file and ES files contains only one generated file.

File: deploy/operator.yaml (hunk 1 above)
Comment (jpkrohling) [reply deploy/operator.yaml]: The port 60000 didn't seem to be used and is probably a left-over from a copy/paste or generated code.

The port 8383 is the one used in newly created operators.

**Comment by jpkrohling** [comment]
> could you please make sure that self-provisioning of ES cluster works?

This really needs an e2e test... It's on your queue, right?

The ES operator is currently not working for me, but seems to be something unrelated to this PR:

```
time="2019-03-07T13:40:34Z" level=error msg="error syncing key (myproject/elasticsearch): Failed to reconcile Roles and RoleBindings for Elasticsearch cluster: failed to create ClusterRole elasticsearch-metrics: clusterroles.rbac.authorization.k8s.io is forbidden: User \"system:serviceaccount:openshift-logging:elasticsearch-operator\" cannot create clusterroles.rbac.authorization.k8s.io at the cluster scope: no RBAC policy matched"
```

**Comment by jpkrohling** [comment]
> would be better to commit the vendor changes in a separate commit to the code changes

Great idea, will do that!

**Comment by jpkrohling** [comment]
I'm merging, as @pavolloffay tested the self-provision and it works.

**Pull request merged**